hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java

```diff
@@ -430,7 +432,19 @@ private Object doLoad(
 		Object entity = persistenceContextEntry.getEntity();
 
 		if ( entity != null ) {
-			return persistenceContextEntry.isManaged() ? entity : null;
+			if ( persistenceContextEntry.isManaged() ) {
+				final LoadQueryInfluencers influencers = event.getSession().getLoadQueryInfluencers();
+				final boolean hasActiveInfluencers = influencers.getEffectiveEntityGraph().getSemantic() != null
+						|| influencers.hasEnabledFetchProfiles();
+				if ( hasActiveInfluencers ) {
+					// we may need to force initialization of some of the entity's attributes
+					persister.initializeLazyProperties( entity, event.getSession() );
```

Comment:
not sure but probably we can pass the `RootGraphImplementor<?>` or the `EffectiveEntityGraph` to the `initializeLazyProperties`, this avoids to call again `session.getLoadQueryInfluencers();` and
`loadQueryInfluencers.getEffectiveEntityGraph()`